.github/workflows/dvc-diff.yml

```diff
@@ -0,0 +1,95 @@
+# Checks for image diffs in a Pull Request and adds a GitHub comment showing the diff
+name: DVC image diff
+
+on:
+  pull_request:
+    paths:
+      - 'pygmt/tests/baseline/*.png.dvc'
+
+jobs:
+  dvc-diff:
+    name: DVC image diff
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2.3.4
+      with:
+        # fetch all history so that git diff works
+        fetch-depth: 0
+
+    - name: Setup data version control (DVC)
+      uses: iterative/setup-dvc@v1.0.2
+
+    - name: Setup continuous machine learning (CML)
+      uses: iterative/setup-cml@v1.0.0
+
+    # Produce the markdown diff report, which should look like:
+    # ## Summary of changed images
+    #
+    # This is an auto-generated report of images that have changed on the DVC remote
+    #
+    # | Status   | Path                                |
+    # |----------|-------------------------------------|
+    # | added    | pygmt/tests/baseline/test_image.png |
+    - name: Put list of images that were added or changed into report
+      run: |
+        echo -e "## Summary of changed images\n" > report.md
+        echo -e "This is an auto-generated report of images that have changed on the DVC remote\n" >> report.md
+        dvc diff --show-md master HEAD >> report.md
+        cat report.md
+
+    - name: Pull image data from cloud storage
+      run: dvc pull --remote upstream
+
+    - name: Put image diff(s) into report
+      env:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+      id: image-diff
+      run: |
+        # Get just the filename of the changed image from the report
+        tail --lines=+7 report.md | cut --delimiter=' ' --fields=7 > diff_files.txt
```

Comment:
The command doesn't work for deleted and modified files.
```
| Status   | Path                                                        |
|----------|-------------------------------------------------------------|
| added    | pygmt/tests/baseline/test_basemap.png                       |
| deleted  | pygmt/tests/baseline/test_solar_terminators.png             |
| modified | pygmt/tests/baseline/test_logo_on_a_map.png                 |
```
I ran the following command to the above "report.md" file:
```
tail --lines=+3 report.md | cut --delimiter=' ' --fields=7 > diff_files.txt
```
it gives me:
```
pygmt/tests/baseline/test_basemap.png


```

Reply:
It would be better if you can delete a dvc file and update a dvc file in this PR, so that we can know the workflow works for added, deleted and modified images.

Reply:
```
tail --lines=+3 report.md | awk '{print $4}'
```
This command gives me the expected output:
```
pygmt/tests/baseline/test_basemap.png
pygmt/tests/baseline/test_solar_terminators.png
pygmt/tests/baseline/test_logo_on_a_map.png
```
NOTE: you need to change `+3` to `+7`.

Reply:
Thanks for the `awk` command suggestion, really need to pick up more bash scripting skills!

> It would be better if you can delete a dvc file and update a dvc file in this PR, so that we can know the workflow works for added, deleted and modified images.

Probably won't work for deleted images since there's nothing to report. But I'm pretty sure added and modified images will work. I'd prefer to test this in a separate PR so that:

1. We keep this PR small and focused. Modifying too many test images will result in a bigger diff to review.
2. People can start using the dvc-diff action in their PRs. At this point in time, most of the changes will be **adding** new images to dvc so we don't need to worry about deleting/modifying images yet.

Reply:
Sounds good to me.